### ISSUE TITLE
Use-cases & design ideas

### ISSUE THREAD

**Issue by helfer** [apollographql/graphql-prism#1]
@abhiaiyer91 @Urigo here we go!

**Comment by stubailo** [comment]
I'm particularly interested in the case where you have a "main" schema, and you want to add an "auxiliary" schema to that. Some cases:

1. Attach an external API like GitHub or Yelp to my app
2. Use a package/service that defines a user login system, and some GraphQL types for that, as part of my schema

In this case, I'm not sure if the correct architecture is:

```
request -> prism
prism -> my API
prism -> GitHub API
```

or

```
request -> my API (with prism package) -> GitHub API
```

Basically not sure if it should be a totally separate layer or if I should be able to just import types from some other server into an existing API.

**Comment by helfer** [comment]
I think prism should just produce a new schema, in which case it becomes secondary which schema is the main schema. Prism should just allow you to combine multiple schemas into one, creating connections between them. Does that make sense?

**Comment by stubailo** [comment]
Are the schemas remote or local? I guess what I'm talking about here is merging a local schema with a remote one. But I suppose that's not a big concern either, I agree that my case reduces to combining two schemas together.

**Comment by helfer** [comment]
Yeah, I think it could be either. So we'll probably need some concept of local vs. remote if we want to treat them differently. 🤔 To start, we could treat all schemas as remote, and just pretend to fetch from them depending on where it actually lives it would go across the network or not. The fetchers package could come in handy for that.

**Comment by PierBover** [comment]
I don't know what's the scope of prism, but I was thinking on solving this problem by having different files with "models" as if we were talking about an ORM.

Each model would have its piece of GraphQL schema and a resolver. This way it would be easy to also integrate authorization in the resolvers. Then just create a function that composes all the models so that there is a single schema, but the resolvers are easily accessible.

It makes sense to me that the type schema and the type resolver should go together in the same place.

**Comment by benoror** [comment]
fwiw in my particular use-case we have a main schema and a secondary schema with some variants. 

I achieved some DRYness & modularization by doing some concatenation (similarly to [this solution](https://github.com/apollographql/graphql-server/issues/43#issuecomment-236091820)), although I would like if this could be done in a more programmatic/ORMesque way.

**Comment by RodMachado** [comment]
Hey @pcarrier, @benoror and @helfer,
We also had the need to modularize some large schemas so we created the [merge-graphql-schemas package](https://github.com/okgrow/merge-graphql-schemas).

Take a look, I hope it helps you too!

Cheers!
